pyserini/dindex/_base.py

```diff
@@ -0,0 +1,135 @@
+#
+# Pyserini: Python interface to the Anserini IR toolkit built on Lucene
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import faiss
+import torch
+from transformers import AutoModel, AutoTokenizer, BertModel, BertTokenizer, DPRContextEncoder, \
+    DPRContextEncoderTokenizer, \
+    RobertaTokenizer
+
+from pyserini.dsearch import AnceEncoder
+
+
+class DocumentEncoder:
+
+    def encode(self, texts):
+        pass
+
+
+def mean_pooling(last_hidden_state, attention_mask):
```

Comment:
a method interspersed w/ classes is a bit difficult to read... I'd put at the end of file or at beginning?